### PR TITLE
vxWorks: set DEFAULT_MIN_STACK_SIZE to 256K and use min_stack to pass initial stack size to rtpSpawn

### PR DESCRIPTION
**Pull request by BaoshanPang**
vxWorks: set DEFAULT_MIN_STACK_SIZE to 256K and use min_stack to pass initial stack size to rtpSpawn

r? @alexcrichton 
cc @n-salim

### CODE REVIEW

File: src/libstd/sys/vxworks/process/process_vxworks.rs [hunk 1]
```diff
@@ -1,10 +1,13 @@
+// Copyright (c) 2019 Wind River Systems, Inc.
```
Comment (lzutao) [inline src/libstd/sys/vxworks/process/process_vxworks.rs]: License?

Comment (BaoshanPang) [reply src/libstd/sys/vxworks/process/process_vxworks.rs]: @lzutao 
Thanks for review on this.
Yes, we would like to add the License text for every file under vxworks directory. Is this a problem?

Comment (n-salim) [reply src/libstd/sys/vxworks/process/process_vxworks.rs]: The intent is "the usual Rust license", i.e. Wind River is not claiming any proprietary rights here. But I'm not a lawyer, let me check with someone who can give a better answer and get back to you.

**Comment by alexcrichton** [comment]
Looks good to me, thanks @BaoshanPang!

Sorry I wasn't super vigilant earlier about this, so `src/libstd/sys/vxworks/fast_thread_local.rs` already has this license header as well, but we would ideally prefer to avoid these license headers if possible. We recently [had a long discussion](https://github.com/rust-lang/rust/pull/62309#issuecomment-515065383) on a separate thread about this as well. If you can remove the license headers that would be great, can you check if you can do that? If you cannot, we will need to consult some lawyers most likely about this and it may take a significant amount of time to land this.

**Comment by n-salim** [comment]
@alexcrichton  Thanks for the feedback. I've asked our IP expert whether we can remove these copyright notices for submissions to the Rust project, and will get back to you with an answer ASAP.

**Comment by n-salim** [comment]
@alexcrichton 

First, let's distinguish between the copyright notice, e.g:

`// Copyright 2019, The Rust Project Developers.`

and the license, e.g.:

`// Licensed under the Apache License, Version 2.0 `

Bearing that distinction in mind, here's the feedback from our IP expert (who oversees Wind River contributions to a number of open source projects).

He made the following very good point:

> GCC, LLVM, The Linux Kernel as a matter of policy - all require a license notice each file – for a very good reason. If a file does not include a licensing information, it is not open source. That is, one is not granted the rights to copy, modified or distribute the file. The number one threat to open source movement is the lack of clear licensing information in a file. A single license.txt file at the top of a project directory is NO guarantee that each file in the subdirectories (or even the same directory) is under the same terms. 

and requested the following:

> Can we request the following notice for WR created files. Note the code will be copyright WR regardless of whether or not we include the notice.

```
// Copyright 2019, The Rust Project Developers.
// Copyright 2019, Wind River Systems
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option.

```
Are you ok with this wording? The license text above  makes it explicit that this file follows the standard Rust licensing policy.

Note that there are a number of existing files in the Rust project that also include explicit 3rd party copyright notices, e.g.:

https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/memchr.rs
https://github.com/rust-lang/rust/blob/master/src/libserialize/json.rs

 In addition the existing Rust Copyright policy (https://github.com/rust-lang/rust/blob/master/COPYRIGHT) policy appears to cover and allow this standard case:

> Copyrights in the Rust project are retained by their contributors. No
> copyright assignment is required to contribute to the Rust project.
> 
> Some files include explicit copyright notices and/or license notices.
> For full authorship information, see the version control history or
> https://thanks.rust-lang.org
> 
> Except as otherwise noted (below and/or in individual files), Rust is
> licensed under the Apache License, Version 2.0 <LICENSE-APACHE> or
> <http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
> <LICENSE-MIT> or <http://opensource.org/licenses/MIT>, at your option.

**Comment by alexcrichton** [comment]
I do not personally have thoughts on this, I will forward this to others who can take care of this.

**Comment by alexcrichton** [comment]
Ok, I'm going to transfer review to...

r? @skade

**Comment by n-salim** [comment]
@alexcrichton 

We've really appreciated the rapid pace at our contributions have been reviewed and accepted up to now, and want to avoid introducing any road blocks.

With the agreement of our IP guy, we've decided to split this into two PRs, one for the "content", and another for the proposed copyright notice.

(1) We'll remove the Wind River copyright notices from this PR, as well as src/libstd/sys/vxworks/fast_thread_local.rs and any other existing affected files. Hopefully that will allow this PR to get back on track.

(2) We'll create a separate PR to propose copyright/license notices for a few of the WR added files that we think need them. Any discussion around such notices can occur in that PR, without any pressure to "hurry". 

Sound good?

**Comment by alexcrichton** [comment]
@n-salim sounds like a good plan to me!

**Comment by skade** [comment]
@n-salim Also sounds great from me, please ping me on the next PR and I'm pretty sure we will get that through quickly!

**Comment by BaoshanPang** [comment]
Hi All,

The Wind River copyritht notices have been removed.

Thanks,
Baoshan

**Pull request merged**